mdtraj/formats/gro.py

```diff
@@ -301,7 +302,9 @@ def _read_topology(self):
                 (thisresnum, thisresname, thisatomname, thisatomnum) = \
                     [line[i*5:i*5+5].strip() for i in range(4)]
                 thisresnum, thisatomnum = map(int, (thisresnum, thisatomnum))
-                if residue is None or residue.resSeq != thisresnum:
+                if residue is None or residue.resSeq != thisresnum or residue.name != thisresname:
```

Comment:
if the residue name changes but the resseq doesn't the residue is changed and a warning (similar to the one of the pdb parser) is thrown